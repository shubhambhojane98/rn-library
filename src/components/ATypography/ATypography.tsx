import React from 'react';
import {StyleProp, StyleSheet, Text, TextProps, TextStyle} from 'react-native';
import {moderateScale} from 'react-native-size-matters';
import {fontFamilies} from '../../constants/fontFamilies';
import {TextAlignment, TypographyVariant} from './ATypographyEnum';
import {defaultScale} from '../../utils/Common';
import {withTheme, useTheme} from '../../core/theming';
import type {Theme} from '../../utils/types';
interface OwnProps {
  variant?: TypographyVariant;
  color?: string;
  style?: StyleProp<TextStyle>;
  textAlign?: TextAlignment;
  fontSize?: number;
  theme: Theme;
}

const defaultFontSize = moderateScale(14, defaultScale);

type Props = OwnProps & TextProps;
const Typography: React.FC<Props> = ({
  variant = TypographyVariant.PRIMARY,
  color,
  style,
  textAlign = TextAlignment.Left,
  children,
  fontSize = defaultFontSize,
  ...otherProps
}) => {
  const {colors} = useTheme();
  return (
    <Text
      style={[
        styles({
          fontSize,
        })[variant],
        {color: color ? color : colors.textColor},
        {textAlign},
        style,
      ]}
      {...otherProps}>
      {children}
    </Text>
  );
};

export default withTheme(Typography);

const styles = (props: {fontSize: number}) =>
  StyleSheet.create({
    primary: {
      fontFamily: `${fontFamilies.Fonts.primary}-Regular`,
      fontSize: moderateScale(props.fontSize, defaultScale),
      fontWeight: 'normal',
    },
    primarySemiBold: {
      fontFamily: `${fontFamilies.Fonts.primary}-SemiBold`,
      fontSize: moderateScale(props.fontSize, defaultScale),
      fontWeight: '600',
    },
    primaryBold: {
      fontFamily: `${fontFamilies.Fonts.primary}-Bold`,
      fontSize: moderateScale(props.fontSize, defaultScale),
      fontWeight: 'bold',
    },
    secondary: {
      fontFamily: `${fontFamilies.Fonts.secondary}-Regular`,
      fontSize: moderateScale(props.fontSize, defaultScale),
      fontWeight: 'normal',
    },
    secondaryDemi: {
      fontFamily: `${fontFamilies.Fonts.secondary}-Demi`,
      fontSize: moderateScale(props.fontSize, defaultScale),
      fontWeight: '600',
    },
    secondaryBold: {
      fontFamily: `${fontFamilies.Fonts.secondary}-Bold`,
      fontSize: moderateScale(props.fontSize, defaultScale),
      fontWeight: 'bold',
    },
  });
