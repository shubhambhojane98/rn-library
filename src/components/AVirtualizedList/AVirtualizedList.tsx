import React from 'react';
import {FlatList} from 'react-native';

const AVirtualizedList: React.FC = ({children}) => {
  return (
    <FlatList
      data={[]}
      showsVerticalScrollIndicator={false}
      keyExtractor={(item, index) => '' + index}
      renderItem={null}
      bounces={false}
      ListHeaderComponent={<>{children}</>}
    />
  );
};

export default AVirtualizedList;
