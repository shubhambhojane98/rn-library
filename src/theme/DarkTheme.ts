import type {Theme} from '../utils/types';

let DarkTheme: Theme;

DarkTheme = {
  dark: true,
  colors: {
    primary: '#BB86FC',
    primaryVariant: '#3700B3',
    secondary: '#03DAC6',
    secondaryVariant: '#018786',
    background: '#262833',
    error: '#CF6679',
    textColor: '#FFF',
    shadowColor: '#D7D7D7',
    black: '#000',
    white: '#FFF',
    lightgrey: '#D7D7D7',
  },
};

export default DarkTheme;
