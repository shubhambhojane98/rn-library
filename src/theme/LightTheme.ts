import type {Theme} from '../utils/types';

let LightTheme: Theme;

LightTheme = {
  dark: false,
  colors: {
    primary: '#6200EE',
    primaryVariant: '#3700B3',
    secondary: '#03DAC6',
    secondaryVariant: '#018786',
    background: '#FFF',
    error: '#B00020',
    textColor: '#000',
    shadowColor: '#222',
    black: '#000',
    white: '#FFF',
    lightgrey: '#D7D7D7',
    backgroundLayout: 'rgba(0, 0, 0, 0.8)',
    transparent: 'transparent',
    linkColor: '#0000EE',
    greyOpac50: 'rgba(53, 53, 53, 0.5)',
    greyOpac10: 'rgba(53, 53, 53, 0.1)',
  },
};

export default LightTheme;
