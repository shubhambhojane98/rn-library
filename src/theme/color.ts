import {Appearance} from 'react-native';

interface colors {
  yellow: string;
  black: string;
  red: string;
  violet: string;
  taupe: string;
  beige: string;
  grey: string;
  lightgrey: string;
  blue: string;
  darkGrey: string;
  white: string;
  transportation: string;
  logistic: string;
  environment: string;
  sports: string;
  healthcare: string;
  background: string;
  textColor: string;
  placeholderTextColor: string;
  switchBackgroundColor: string;
  paginationDotColor: string;
  paginationDotBgColor: string;
  blackOpac80: string;
  blackOpac60: string;
  blackOpac40: string;
  blackOpac20: string;
  greyOpac80: string;
  greyOpac60: string;
  greyOpac50: string;
  greyOpac40: string;
  greyOpac20: string;
  greyOpac10: string;
  switchActiveBgColor: string;
  switchInActiveBgColor: string;
  switchKnobColor: string;
}

const primaryColors = {
  yellow: '#FAD000',
  black: '#231F20',
  red: '#A81E23',
  violet: '#533467',
  taupe: '#695958',
  beige: '#C3B59F',
  grey: '#353535',
  white: '#fff',
  lightgrey: '#D7D7D7',
  blue: '#007BC1',
  darkGrey: 'rgba(0, 0, 0, 0.2)',
};

const opacColors = {
  blackOpac80: 'rgba(0, 0, 0, 0.8)',
  blackOpac60: 'rgba(0, 0, 0, 0.6)',
  blackOpac40: 'rgba(0, 0, 0, 0.4)',
  blackOpac20: 'rgba(0, 0, 0, 0.2)',
  greyOpac80: 'rgba(53, 53, 53, 0.8)',
  greyOpac60: 'rgba(53, 53, 53, 0.6)',
  greyOpac50: 'rgba(53, 53, 53, 0.5)',
  greyOpac40: 'rgba(53, 53, 53, 0.4)',
  greyOpac20: 'rgba(53, 53, 53, 0.2)',
  greyOpac10: 'rgba(53, 53, 53, 0.1)',
};

const categoryColors = {
  transportation: '#007BC1',
  logistic: '#EF4065',
  environment: '#009A46',
  sports: '#F26524',
  healthcare: '#00ABBE',
};

let themeColors: colors;

const colorScheme = Appearance.getColorScheme();
if (colorScheme === 'dark') {
  // Use dark color scheme
  themeColors = {
    ...primaryColors,
    ...categoryColors,
    ...opacColors,
    background: primaryColors.black,
    textColor: primaryColors.white,
    placeholderTextColor: '#5D5D5D',
    switchBackgroundColor: primaryColors.white,
    paginationDotColor: primaryColors.white,
    paginationDotBgColor: primaryColors.darkGrey,
    switchActiveBgColor: primaryColors.yellow,
    switchInActiveBgColor: primaryColors.white,
    switchKnobColor: primaryColors.black,
  };
} else {
  // Use light color scheme
  themeColors = {
    ...primaryColors,
    ...categoryColors,
    ...opacColors,
    background: primaryColors.yellow,
    textColor: primaryColors.black,
    placeholderTextColor: '#5D5D5D',
    switchBackgroundColor: primaryColors.lightgrey,
    paginationDotColor: primaryColors.white,
    paginationDotBgColor: primaryColors.darkGrey,
    switchActiveBgColor: primaryColors.yellow,
    switchInActiveBgColor: primaryColors.white,
    switchKnobColor: primaryColors.black,
  };
}

export default themeColors;
