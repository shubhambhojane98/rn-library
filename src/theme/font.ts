import { Platform, PlatformOSType } from 'react-native';
import { Fonts } from '../utils/types';


export default function configureFonts(config?: {
    [platform in PlatformOSType | 'default']?: Fonts;
  }): Fonts {
    const fonts = Platform.select({ ...config }) as Fonts;
    return fonts;
  }