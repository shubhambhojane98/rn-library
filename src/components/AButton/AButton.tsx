import React from 'react';
import {
  TouchableOpacity,
  StyleSheet,
  StyleProp,
  TextStyle,
  ImageBackground,
  ImageStyle,
  ActivityIndicator,
} from 'react-native';
import FastImage, {Source} from 'react-native-fast-image';
import {moderateScale} from 'react-native-size-matters';
import IconSVG from '../../assets/svgs';
import {fontFamilies} from '../../constants/fontFamilies';
import {defaultScale} from '../../utils/Common';
import ATypography from '../ATypography/ATypography';
import {TypographyVariant} from '../ATypography/ATypographyEnum';
import {withTheme, useTheme} from '../../core/theming';
import type {Theme} from '../../utils/types';

interface Props {
  backgroundColor?: string;
  width?: number;
  height?: number;
  title?: string;
  onPress: () => void;
  borderRadius?: number;
  margin?: number;
  imageStyle?: StyleProp<ImageStyle>;
  backgroundSource?: any;
  padding?: number;
  iconName?: any;
  iconHeight?: number;
  iconWidth?: number;
  sourceImage?: number | Source;
  color?: any;
  fontFamily?: string;
  fontSize?: number;
  disabled?: boolean;
  textStyle?: StyleProp<TextStyle>;
  textAlign?: any;
  fontWeight?: any;
  textDecorationLine?: any;
  hyperlink?: boolean;
  borderWidth?: number;
  borderColor?: string;
  isLoading?: boolean;
  theme: Theme;
}
const AButton = (props: Props) => {
  const {
    title,
    onPress,
    backgroundColor,
    width,
    height,
    margin,
    fontFamily,
    fontSize,
    textAlign = 'center',
    sourceImage,
    padding,
    backgroundSource,
    color,
    borderRadius,
    imageStyle = {borderRadius, padding},
    iconName,
    iconHeight,
    iconWidth,
    fontWeight,
    textDecorationLine,
    hyperlink,
    borderWidth,
    borderColor,
    isLoading = false,
  } = props;

  const {colors} = useTheme();
  const bgColor = backgroundColor ? backgroundColor : colors.primary;

  const stylesWithProp = styles({colors});

  return (
    <TouchableOpacity
      onPress={onPress}
      style={{
        ...stylesWithProp.container,
        height,
        width,
        backgroundColor: hyperlink ? colors.primaryVariant : bgColor,
        margin,
        borderRadius,
        borderWidth,
        borderColor,
      }}>
      {backgroundSource ? (
        <ImageBackground
          source={backgroundSource}
          resizeMode="cover"
          imageStyle={imageStyle}
          style={{
            ...stylesWithProp.container,
            margin,
            flex: 1,
            height: '100%',
          }}
        />
      ) : null}

      {iconName ? (
        <IconSVG name={iconName} height={iconHeight} width={iconWidth} />
      ) : null}

      {!hyperlink && (
        <ATypography
          children={title}
          variant={TypographyVariant.PRIMARY_BOLD}
          color={color || colors.textColor}
          style={{
            ...stylesWithProp.text,
            textAlign,
            fontFamily,
            fontSize,
            fontWeight,
            textDecorationLine,
          }}
        />
      )}
      {hyperlink && (
        <ATypography
          children={title}
          variant={TypographyVariant.SECONDARY_DEMI}
          style={{
            ...stylesWithProp.hyperlinkText,
            textAlign,
            fontFamily,
            fontSize,
            fontWeight,
            textDecorationLine,
          }}
        />
      )}
      {sourceImage ? (
        <FastImage
          style={{...stylesWithProp.container, height, width}}
          source={sourceImage ? sourceImage : 0}
          resizeMode={FastImage.resizeMode.contain}
        />
      ) : null}
      {isLoading ? <ActivityIndicator size={'small'} /> : null}
    </TouchableOpacity>
  );
};
const styles = (props: {colors: any}) =>
  StyleSheet.create({
    container: {
      alignItems: 'center',
      flexDirection: 'row',
      textAlign: 'center',
      justifyContent: 'center',
    },
    text: {
      alignItems: 'center',
      fontSize: moderateScale(18, defaultScale),
      fontFamily: fontFamilies.Fonts.primary,
      padding: moderateScale(10, defaultScale),
    },
    hyperlinkText: {
      alignItems: 'center',
      fontSize: moderateScale(18, defaultScale),
      fontFamily: fontFamilies.Fonts.primary,
      padding: moderateScale(10, defaultScale),
      color: props.colors.textColor,
    },
  });
export default withTheme(AButton);
