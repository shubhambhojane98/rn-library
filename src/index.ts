import {defaultScale} from './utils/Common';
import {consoleLog, consoleWarn, consoleInfo} from './utils/Log';
import {CheckboxStatus} from './components/ACheckbox/ACheckboxEnum';
import {Color} from './theme';
export {default as AButton} from './components/AButton/AButton';
export {default as ATypography} from './components/ATypography/ATypography';
export {default as AAvatar} from './components/AAvatar/AAvatar';
export {default as ABadge} from './components/ABadge/ABadge';
export {default as AImage} from './components/AImage/AImage';
export {default as ARadioButton} from './components/ARadioButton/ARadioButton';
export {default as ARadioButtonGroup} from './components/ARadioButtonGroup/ARadioButtonGroup';
export {default as ATextInput} from './components/ATextInput/ATextInput';
export {default as ASwitch} from './components/ASwitch/ASwitch';
export {default as AVirtualizedList} from './components/AVirtualizedList/AVirtualizedList';
export {default as AModal} from './components/AModal/AModal';
export {default as ASwitchSelector} from './components/ASwitchSelector/ASwitchSelector';
export {default as AStarRating} from './components/AStarRating/AStarRating';
export {default as ACheckbox} from './components/ACheckbox/ACheckbox';
export {default as AToast} from './components/AToast/AToast';
export {default as ACard} from './components/ACard/ACard';
export {consoleLog, consoleWarn, consoleInfo};
export {Color};
export {CheckboxStatus};
