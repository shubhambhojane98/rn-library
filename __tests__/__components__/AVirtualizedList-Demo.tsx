import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import AVirtualizedList from '../../src/components/AVirtualizedList/AVirtualizedList';
import AButton from '../../src/components/AButton/AButton';
import ATypography from '../../src/components/ATypography/ATypography';
import {Color} from '../../src/theme';
import {Alert} from 'react-native';

it('renders correctly with default prop', () => {
  const tree = renderer
    .create(
      <AVirtualizedList>
        <ATypography fontSize={12}>{'Virutalized List'}</ATypography>
        <AButton
          title={'Simple Button'}
          onPress={() => Alert.alert('button Pressed')}
          width={290}
          color={Color.blue}
          borderRadius={49}
        />
      </AVirtualizedList>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
