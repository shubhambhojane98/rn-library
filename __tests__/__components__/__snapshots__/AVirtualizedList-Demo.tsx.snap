// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with default prop 1`] = `
<RCTScrollView
  ListHeaderComponent={
    <React.Fragment>
      <Typography
        fontSize={12}
      >
        Virutalized List
      </Typography>
      <AButton
        borderRadius={49}
        color="#007BC1"
        fontSize={22.114285714285714}
        height={61.42857142857143}
        onPress={[Function]}
        title="Simple Button"
        width={290}
      />
    </React.Fragment>
  }
  bounces={false}
  data={Array []}
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  showsVerticalScrollIndicator={false}
  stickyHeaderIndices={Array []}
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Mulish-Regular",
              "fontSize": 14.742857142857144,
              "fontWeight": "normal",
            },
            Object {
              "color": "#231F20",
            },
            Object {
              "textAlign": "left",
            },
            undefined,
          ]
        }
      >
        Virutalized List
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": undefined,
            "borderColor": undefined,
            "borderRadius": 49,
            "borderWidth": undefined,
            "flexDirection": "row",
            "height": 61.42857142857143,
            "justifyContent": "center",
            "margin": undefined,
            "opacity": 1,
            "textAlign": "center",
            "width": 290,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Mulish-Bold",
                "fontSize": 21.13142857142857,
                "fontWeight": "bold",
              },
              Object {
                "color": "#231F20",
              },
              Object {
                "textAlign": "left",
              },
              Object {
                "alignItems": "center",
                "color": "#007BC1",
                "fontFamily": undefined,
                "fontSize": 22.114285714285714,
                "fontWeight": undefined,
                "padding": 12.285714285714285,
                "textAlign": "center",
                "textDecorationLine": undefined,
              },
            ]
          }
        >
          Simple Button
        </Text>
        <ActivityIndicator
          size="small"
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;
